CMakeLists.txt

```diff
@@ -129,15 +130,22 @@ if("${CMAKE_EXECUTABLE_SUFFIX}" STREQUAL "")
   set(CMAKE_EXECUTABLE_SUFFIX ".x")
 endif()
 
-add_subdirectory(protos)
 
-# ###################################
+####################################
 # - Post dependencies setup --------
 include(cmake/setup_compiler.cmake)
 
+# Setup code coverage components
+include(cmake/setup_coverage.cmake)
+
 # Setup IWYU if enabled
 include(cmake/setup_iwyu.cmake)
 
+####################################
+# - Configure protobufs
+add_subdirectory(protos)
+
+
```

Comment:
It was move to be past the code coverage setup so that the coverage flags could be attached to the protobuf libs, which I put with the compiler and IWYU configuration because it seemed similar.

Reply:
The problem with this is it puts the proto code after `setup_compiler` which adds debugging flags and `clang-tidy`. Things we dont need with external or auto generated code.